ROS/controls/src/moteus_wrapper.py

```diff
@@ -0,0 +1,213 @@
+    fixed number of motors; you cannot change the number (or identifiers)
+    of the motors that are serviced after you instantiate the class.
+    Instead, create a new instance of the class and replace the
+    old instance to add more motors
+
+    NOTE: The class is intended to function as a singleton, as
+    each instance serves as a publisher to the 'current_orientation' topic.
+
+    NOTE: This class is not thread-safe. You must ensure that concurrent
+    accesses are properly synchronized
+
+    ----------
+    Attributes
+    ----------
+    hw_refresh_rate : float
+        The rate, in Hertz, at which readings of the Moteus
+        motors for their current respective statuses are taken
+
+    num_motors : int
+        The number of motors that are controller
+
+    id_map : { int : int }
+        Maps the hardware ids given to each motor to the
+        CAN bus lane which that motor communicates on
+
+    -------
+    Methods
+    -------
+    run_motor_orientation_loop():
+        Begins an infinite loop within which the instance publishes
+        on the ROS topic about the current orientation of the motors
+
+    shutdown_moteus():
+        Stops the execution of the node and performs any necessary
+        resource cleanup and/or hardware cleanup
+    """
+
+    # MARK: - Constructors -
+
+    def __init__(self, motor_list, hw_refresh_rate: float = constants.CONTROLS_PKG_DEFAULT_HW_REFRESH_RATE):
+        """Construct a new wrapper around the Moteus class
+
+        @param motor_list: A list which maps each motor id to its
+        respective CAN bus lane
+        
+        @param hw_refresh_rate: The rate, in Hertz, at which readings of the Moteus
+        motors for their current respective statuses are taken
+        """
+        if len(motor_list) == 0:
+            raise ValueError('The wrapper class expects at least one motor')
+
+        self.hw_refresh_rate = hw_refresh_rate
+        self.num_motors = len(motor_list)
+        self.id_map = motor_list
+
+        # The arrangement of motor ids into CAN lanes can be computed
+        # by processing the mapping of motors IDs to the CAN bus lanes
+        moteus_ids = []
+        current_max_lane_id = -1
+
+        # We build all of data from the mapping at once
+        # instead of using list/dictionary comprehensions
+        # to reduce the number of iterations through the list
+        for motor_id, can_bus_lane_id in motor_list.items():
+            # Determines if we need to add more lanes
+            if can_bus_lane_id > current_max_lane_id:
+                # Add (can_bus_lane_id - current_max_lane_id) more lanes
+                moteus_ids += [[] for _ in range(current_max_lane_id, can_bus_lane_id)]
+
+            moteus_ids[can_bus_lane_id].append(motor_id)
+
+        # Note that the call to the initializer will block
+        self.moteus_instance = Moteus(moteus_ids, simulation=True)
+
+        self.desired_orientation_sub = rospy.Subscriber(
```

Comment:
It seems like the publisher and subscriber are talking on the same topic. We need one input to be from the IK and the output to be the current motor positions, if they are on the same topic I believe this will just cause the motors to go their current position?

Reply:
They are using different topics. [See here for details](https://github.com/Northeastern-University-Robotics/Pen-Plotter/blob/1ca44d1d6066bfa1c304b1a85e38048aa746549d/ROS/controls/src/moteus_wrapper.py#L99-L110) and note that `constants.CONTROLS_PKG_DESIRED_ORIENTATION_TOPIC` is not the same as `constants.CONTROLS_PKG_CURRENT_ORIENTATION_TOPIC`

Reply:
Sorry my bad I didn't see the word desired lol.